mxcube3/routes/SampleCentring.py

```diff
@@ -264,8 +294,16 @@ def update_shapes():
         # Shape does not have any refs, create a new Centered position
         if not refs:
             x, y = shape_data["screen_coord"]
-            mpos = mxcube.diffractometer.get_centred_point_from_coord(x, y)
-            shape = mxcube.shapes.add_shape_from_mpos([mpos], (x, y), t)
+            mpos = mxcube.diffractometer.get_centred_point_from_coord(x, y, return_by_names=True)
+            pos.append(mpos)
+            # MD3 needs the center of the grid instead of the top left corner
```

Comment:
I do not remove the top-left corner, just adding an extra field for the center of the grid. Which is then applied in qutils.py L688 `acq.acquisition_parameters.centred_position = grid.get_centred_position()`. I do not know about other diffs but ours needs the center position, and the top-left pos is needed for the client.